Python 3.13 support: _Py_IsFinalizing() has been removed from Python 3.13.

**Describe the bug**

While compiling: 

```
FAILED: Code/RDBoost/Wrap/CMakeFiles/rdBase.dir/RDBase.cpp.o
    /opt/rh/gcc-toolset-12/root/usr/bin/c++ -DBOOST_ALL_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_SERIALIZATION_DYN_LINK -DBOOST_STACKTRACE_ADDR2LINE_LOCATION=\"/usr/bin/addr2line\" -DBOOST_STACKTRACE_USE_ADDR2LINE -DBOOST_STACKTRACE_USE_BACKTRACE -DBOOST_STACKTRACE_USE_NOOP -DRDKIT_DYN_LINK -DRDK_64BIT_BUILD -DRDK_BUILD_COORDGEN_SUPPORT -DRDK_BUILD_DESCRIPTORS3D -DRDK_BUILD_MAEPARSER_SUPPORT -DRDK_BUILD_THREADSAFE_SSS -DRDK_BUILD_XYZ2MOL_SUPPORT -DRDK_BUILD_YAEHMOP_SUPPORT -DRDK_HAS_EIGEN3 -DRDK_TEST_MULTITHREADED -DRDK_USE_BOOST_IOSTREAMS -DRDK_USE_BOOST_SERIALIZATION -DRDK_USE_STRICT_ROTOR_DEFINITION -DRDK_USE_URF -D_GLIBCXX_USE_CXX11_ABI=0 -DrdBase_EXPORTS -I/project/build/temp.linux-x86_64-cpython-313/rdkit/External -I/project/build/temp.linux-x86_64-cpython-313/rdkit/External/RingFamilies/RingDecomposerLib/src/RingDecomposerLib -I/opt/python/cp313-cp313/include/python3.13 -I/tmp/pip-build-env-o5feru76/overlay/lib/python3.13/site-packages/numpy/_core/include -I/project/build/temp.linux-x86_64-cpython-313/rdkit/Code -isystem /root/.conan/data/boost/1.85.0/chris/mod_boost/package/b5c800decfecf8411fb7deafae77548a09bfd890/include -isystem /usr/include/eigen3 -isystem /root/.conan/data/bzip2/1.0.8/_/_/package/3cfc45772763dad1237052f26c1fe8b2bae3f7d2/include -isystem /root/.conan/data/zlib/1.3.1/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -isystem /root/.conan/data/libbacktrace/cci.20210118/_/_/package/2a19826344ff00be1c04403f2f8e7008ed3a7cc6/include -m64 -mpopcnt -Wno-deprecated -Wno-unused-function -fno-strict-aliasing -Wall -Wextra -fPIC -O3 -DNDEBUG -std=gnu++17 -fPIC -MD -MT Code/RDBoost/Wrap/CMakeFiles/rdBase.dir/RDBase.cpp.o -MF Code/RDBoost/Wrap/CMakeFiles/rdBase.dir/RDBase.cpp.o.d -o Code/RDBoost/Wrap/CMakeFiles/rdBase.dir/RDBase.cpp.o -c /project/build/temp.linux-x86_64-cpython-313/rdkit/Code/RDBoost/Wrap/RDBase.cpp
    /project/build/temp.linux-x86_64-cpython-313/rdkit/Code/RDBoost/Wrap/RDBase.cpp: In destructor ‘virtual PyLogStream::~PyLogStream()’:
    /project/build/temp.linux-x86_64-cpython-313/rdkit/Code/RDBoost/Wrap/RDBase.cpp:81:10: error: ‘_Py_IsFinalizing’ was not declared in this scope; did you mean ‘Py_IsFinalizing’?
       81 |     if (!_Py_IsFinalizing()) {
          |          ^~~~~~~~~~~~~~~~
          |          Py_IsFinalizing
```


_Py_IsFinalizing() has been removed from Python 3.13.  See: https://github.com/python/cpython/issues/108014


https://github.com/rdkit/rdkit/blob/67f507a15c69616332c97234a8c96b016bad0d4d/Code/RDBoost/Wrap/RDBase.cpp#L81


**To Reproduce**
Compile RDKit for Python 3.13. The cp313 builds all fail with this error at  https://github.com/kuelumbus/rdkit-pypi/actions/runs/10466978307

**Configuration (please complete the following information):**
 - RDKit version: Release_2024_03_5
 - OS: all
 - Python version (if relevant): python 3.13
 - Are you using conda?
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit?

--- a/Code/RDBoost/Wrap/RDBase.cpp
+++ b/Code/RDBoost/Wrap/RDBase.cpp
@@ -81,7 +81,7 @@ struct PyLogStream : std::ostream, std::streambuf {
 #if PY_VERSION_HEX < 0x30d0000
     if (!_Py_IsFinalizing()) {
 #else
-    if (Py_IsFinalizing()) {
+    if (!Py_IsFinalizing()) {
 #endif
       Py_XDECREF(logfn);
     }

--- a/Code/RDBoost/Wrap/RDBase.cpp
+++ b/Code/RDBoost/Wrap/RDBase.cpp
@@ -78,7 +78,11 @@ struct PyLogStream : std::ostream, std::streambuf {
   }
 
   ~PyLogStream() {
+#if PY_VERSION_HEX < 0x30d0000
     if (!_Py_IsFinalizing()) {
+#else
+    if (Py_IsFinalizing()) {
+#endif
       Py_XDECREF(logfn);
     }
   }

